### DISCUSSION TITLE
Wrong Eth balance returned in contract for <ADDR>.balance

### DISCUSSION THREAD

**Discussion by saad-s** [ConsenSys-archive/truffle#5373] [General Q&A]
I am trying to read eth balance of a given address within contract. But the value returned is too large. 

My test setup is `MAC OS 12.4 (intel)` and 
```bash 
truffle(development)> version
Truffle v5.5.21 (core: 5.5.21)
Ganache v7.2.0
Solidity - 0.8.15 (solc-js)
Node v18.2.0
Web3.js v1.7.4
```
Contract function 
```solidity 
/**
* @notice returns balance of given address in wei
*/
function getEthBalance(address addr) public view returns (uint) {
  return addr.balance;
}
```
sample test case
```JS
// --- snip ---
    it("should return ethereum balance of given address", async () => {
      // read balance by calling contract function 
      const balanceFromCall = new BN(await this.ethStrategy.getEthBalance(deployer));
      // read balance using web3 
      const balanceFromWeb3 = new BN(await web3.eth.getBalance(deployer));
      expect(balanceFromWeb3).to.be.bignumber.equal(balanceFromCall);
    });
// --- snip ---
```
Balance value in `balanceFromWeb3` is `97.01` ethers which is correct value and matches Ganache UI value for account as well. 
But balance value from contract in `balanceFromWeb3` is `180143888.x` ethers.

I found this issue in geth [(link)](https://ethereum.stackexchange.com/questions/87754/msg-sender-balance-showing-incorrect-amount) which seems related but I am unsure what to change in my setup to fix it.

**Comment by apdan510** [comment]
@saad-s 
Your Ganache is updated no need to make changes to your node. Any chance you can share your full repo? Would love to see the other configs, other files, etc. 

Kind regards,
Alex | Truffle Support
